src/main/java/PersonalBudget/business/balance/domain/service/BalancePageHandler.java

```diff
@@ -0,0 +1,96 @@
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static PersonalBudget.common.util.Date.getFirstDayCurrentMonth;
+import static PersonalBudget.common.util.Date.getFirstDayCurrentYear;
+import static PersonalBudget.common.util.Date.getFirstDayPreviousMonth;
+import static PersonalBudget.common.util.Date.getLastDayCurrentMonth;
+import static PersonalBudget.common.util.Date.getLastDayCurrentYear;
+import static PersonalBudget.common.util.Date.getLastDayPreviousMonth;
+
+@RequiredArgsConstructor
+@Service
+public class BalancePageHandler {
+
+    private final BalanceGateway balanceGateway;
+    private final BalanceTemplateService balanceTemplateService;
+    private static final String BALANCE_PAGE = "menu/balance";
+    private static final String MENU_PAGE = "menu/main";
+
+
+    public String handleCurrentMonthBalancePage(Model model) {
+        LocalDate dateFrom = getFirstDayCurrentMonth();
+        LocalDate dateTo= getLastDayCurrentMonth();
+        handleBalancePage(model, dateFrom, dateTo);
+        return BALANCE_PAGE;
+    }
+
+    public String handlePreviousMonthBalancePage(Model model) {
+        LocalDate dateFrom = getFirstDayPreviousMonth();
+        LocalDate dateTo = getLastDayPreviousMonth();
+        handleBalancePage(model, dateFrom, dateTo);
+        return BALANCE_PAGE;
+    }
+
+    public String handleCurrentYearBalancePage(Model model) {
+        LocalDate dateFrom = getFirstDayCurrentYear();
+        LocalDate dateTo = getLastDayCurrentYear();
+        handleBalancePage(model, dateFrom, dateTo);
+        return BALANCE_PAGE;
+    }
+
+    public String handleNonstandardBalancePage(NonstandardDateDTO nonstandardDateDTO, Model model) {
+        LocalDate dateFrom = nonstandardDateDTO.dateFrom();
+        LocalDate dateTo = nonstandardDateDTO.dateTo();
+        if(dateTo.isBefore(dateFrom)) {
+            balanceTemplateService.addTwoDatesComparisonAttribute(model);
+            return MENU_PAGE;
+        }
+        handleBalancePage(model, dateFrom, dateTo);
+        return BALANCE_PAGE;
+    }
+
+    public String handleNonstandardBalancePageWithoutDateRange(Model model) {
+        balanceTemplateService.addNoDateRangeAttribute(model);
+        return MENU_PAGE;
+    }
+
+    public void handleBalancePage(Model model, LocalDate dateFrom, LocalDate dateTo) {
+        List<CategorySumDTO> incomeCategoriesSum  = balanceGateway.fetchUserIncomeCategoriesSums(dateFrom, dateTo);
+        List<CategorySumDTO> expenseCategoriesSum = balanceGateway.fetchUserExpenseCategoriesSums(dateFrom, dateTo);
+        List<List<Object>> incomeChartData = mapToChartData(incomeCategoriesSum);
+        List<List<Object>> expenseChartData = mapToChartData(expenseCategoriesSum);
+        List<ParticularActivityDTO> particularIncomes = balanceGateway.fetchUserParticularsIncomeCategory(dateFrom, dateTo);
+        List<ParticularActivityDTO> particularExpenses = balanceGateway.fetchUserParticularsExpenseCategory(dateFrom, dateTo);
```

Comment:
You make calls to income and expense 2 times, can you reduce the number of calls? If those were calls to external services or a database it would be pretty expensive.